### ISSUE TITLE
[2][wmbutler][4][kapeer42] Integrate proposal into send modal even if user not logged in

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-ui#1536]
**Describe the bug**
Can't send fund if not logged in.

**To Reproduce**
1. Go to https://wallet.bitshares.org/#/account/myaccount
2. Click on send
3. input recipient account
4. input amount
5. "send" button is still disabled

**Expected behavior**
Able to click "send" button, after clicked, if not logged in, show the login window (<del>perhaps fill in user name and lock it</del> Update: due to proposal feature, don't force the same user name, more info: https://github.com/bitshares/bitshares-ui/issues/1285)

**Screenshots**
![image](https://user-images.githubusercontent.com/9946777/40268548-c552612c-5b6f-11e8-928f-01645ea583b8.png)

**Desktop (please complete the following information):**
 - OS: Win7
 - Browser Chrome
 - Version latest

**Comment by wmbutler** [comment]
I gave this 7 hours to account for proposed transactions and testing.

**Comment by abitmore** [comment]
@wmbutler since the original design has been broken (see #1285), IMHO the "proposed transaction" feature need an overall redesign. The original design was: giving users the option to propose before broadcast the transaction (thus there is a "propose" radio button on confirmation modal, although it confuses new users, it works for advanced users). Better discuss / plan it in #1285, otherwise likely we'll waste some efforts here.

**Comment by wmbutler** [comment]
This is probably due to my lack of understanding, but why would we require a user to explicitly propose a transactions. Why not treat it as a proposal if the user doesn't hold the keys?

**Comment by abitmore** [comment]
Creating a proposal or not, should be decided by the user but not an opinionated UI. That said, even the user has the key, he should also be able to propose.

Proposing is a complicated feature, IMHO not good to duplicate it everywhere, but better to make it as a dedicated feature, like transaction confirmation modal.

**Comment by abitmore** [comment]
See more discussions here: https://github.com/bitshares/bitshares-ui/issues/1284#issuecomment-376590841

**Comment by abitmore** [comment]
Actually, the user stories listed above fit all actions (including placing an order, cancelling an order, fund fee pool, issuing asset and etc), but not only limited to transfer. That's why I think it should be a generic feature.

**Comment by wmbutler** [comment]
Is this a true statement?

"I cannot send funds if I do not hold the keys"

If so, I should only see a "Propose" button for situations where I do not hold the keys, correct?

**Comment by abitmore** [comment]
@wmbutler you're correct.

By the way, do we know whether the user hold the keys if she didn't login/unlock?

**Comment by svk31** [comment]
Only for wallet mode, in cloud mode you can't really be sure.

**Comment by wmbutler** [comment]
What if we keep the button labeled as "Send". So the user fills in all the values, they click send. The subsequent dialog offers the user two options:

1. Login to complete the transaction
2. Propose this transaction

We could add a circle question mark to help users understand the nature of the propose transaction.

**Comment by gibbsfromncis** [comment]
@wmbutler @abitmore I understand the issue in general terms but can you please specify requirements and I'll work on it

**Comment by startailcoon** [comment]
@dot5enko do you intend to do work on this? 

It seems that @gibbsfromncis is waiting for some details by @wmbutler or @abitmore here. What is the status of this issue?

**Comment by abitmore** [comment]
@startailcoon I don't think the issue is pending my input. It does need UX.

**Comment by abitmore** [comment]
Hmm, just checked latest UI (develop branch), it seems that the "send" button/menu has disappeared if I'm not logged in. But other menus are there. Not sure why. Can we be a bit more consistent? @startailcoon 

![image](https://user-images.githubusercontent.com/9946777/45263098-a9c64b00-b423-11e8-8d63-cf63b24e650d.png)

Unable to send nor propose a send via the "send(legacy)" menu as well. 
![image](https://user-images.githubusercontent.com/9946777/45263226-b2b81c00-b425-11e8-839c-b18c562f2bd1.png)

**Comment by gibbsfromncis** [comment]
@abitmore what's wrong? Looks like I can do it:

<img width="714" alt="screenshot 2018-09-09 13 46 33" src="https://user-images.githubusercontent.com/35899973/45264075-c6b84980-b436-11e8-924b-4705625752eb.png">

Yep UX is terrible but anyway it's possible

**Comment by abitmore** [comment]
@wmbutler as I've commented above, IMHO we need an overall UX (re)design for creating proposals. Proposals are not only for "send" feature.

**Comment by wmbutler** [comment]
@abitmore but we don't have to solve all the problems at once. If I want to propose a send operation, it makes sense to do that from the send modal.

**Comment by wmbutler** [comment]
Proposed UX

- User clicks Send and gets modal
- Modal has pills at top for Send / Propose
- Send is default
- Add from input field but lock it down to active account for the "Send operation"
- If user toggles to "Propose" operation, unlock the FROM field and incorporate typeahead
- Also change label on Send button to "Propose"

![screen shot 2018-09-14 at 12 43 18 pm](https://user-images.githubusercontent.com/1254810/45566460-576a9d00-b81c-11e8-9e94-8105edd3fc16.png)

**Comment by wmbutler** [comment]
@gibbsfromncis you may want to antify this...

**Comment by kapeer42** [comment]
@dot5enko Was [this](https://github.com/bitshares/bitshares-ui/issues/1536#issuecomment-419663041) a claim?
If not and the UX by @wmbutler is validated I'd like to claim it.

**Comment by gibbsfromncis** [comment]
@startailcoon please give your feedback for @kapeer42 

@kapeer42 if you would take it make you you will use our [style-guide ](https://styleguide.bitshares.org/#/dark) for this window. Do not use old inputs or create your own buttons groups ( Send, Propose ). Also this task is blocked by this one https://github.com/bitshares/bitshares-ui/issues/1861. So you should wait while I finish required UI elements for style-guide and then you can start to work on implementing this one.

**Comment by startailcoon** [comment]
Since there is no reply from @dot5enko, and @kapeer42 posted a claim request I would suggest we can add this to claim the later.

@kapeer42 remember to make sure the transaction confirmation dialog remains consistent with the choices that is made in this section.

**Comment by kapeer42** [comment]
@wmbutler 
What is the asset used if the button in the header drop down is used?

**Comment by kapeer42** [comment]
@gibbsfromncis There is no `textarea` component (for the `memo / message` section) currently in the style-guide.

**Comment by kapeer42** [comment]
@wmbutler What account is used to propose?
What happens when you have multiple accounts in a key file and want to propose with a particular one?

**Comment by wmbutler** [comment]
Looking really good.

- [ ] Please make SEND and PROPOSE pills the same width and center them.
- [ ] Misspelling: Propose to send fund -> Propose to send funds

**Comment by startailcoon** [comment]
This should also be using the new style guide with ANT components

**Comment by wmbutler** [comment]
- [ ] Only allow FROM to be populated if PROPOSE is selected

@startailcoon I think we will make an issue for the ANT redesign.

>What is the asset used if the button to open the modal in the header drop down is used?

Empty field for the asset

>Is it possible to select the asset sent in the send modal?

yes

>If yes, should I use the AmountSelector (used in the current send modal)?

I think yes, but may not understand the question

>If yes again, should I fork it to use the ANT components (@gibbsfromncis)?

We definitely would like to see this as ANT as much as possible

> What account is used to propose?

Use the active account by default and then let the user type an arbitrary name otherwise.
> What happens when you have multiple accounts in a key file and want to propose with a particular one?

@kapeer42 A user should be able to arbitrarily type any FROM account for the proposal.

**Comment by kapeer42** [comment]
@wmbutler 
> A user should be able to arbitrarily type any FROM account for the proposal.

I think you misunderstood me (although your answer was useful), let me rephrase:

Given the resulting operation:
> Account P proposed to send amount X of asset Y from account F to account T

What P is chosen?

**Comment by kapeer42** [comment]
On the ANT side I see 5 subtasks/dependencies:
- [ ] Modify ANT Form.Item to be able to display something on the right of the label (like the two components below)
- [ ] ANTify AccountSelector ![accountselector_current](https://user-images.githubusercontent.com/35779885/46245656-aeae6700-c3f1-11e8-9292-1052071f18dc.png)
- [ ] ANTify AmountSelector ![amountselector_current](https://user-images.githubusercontent.com/35779885/46245660-b40bb180-c3f1-11e8-95e4-ca160ecc16a4.png)
- [ ] Expose ANT Input.TextArea (for the memo)
- [ ] ANTify BaseModal (overlaps with https://github.com/bitshares/bitshares-ui/issues/1664 I think)
- [ ] Expose ANT Button.Group + make it able to have equal width buttons (issue already opened here https://github.com/bitshares/bitshares-ui/issues/1861)

**Comment by startailcoon** [comment]
See #1872 for ANTified components possibly

**Comment by wmbutler** [comment]
@kapeer42 thanks for clarifying. I definitely missed an important piece. Please see this mockup and the change in text to reflect.

![screen shot 2018-10-01 at 11 18 49 am](https://user-images.githubusercontent.com/1254810/46301427-f1944a00-c56b-11e8-9b95-34939fadfc6c.png)

**Comment by kapeer42** [comment]
See https://github.com/bitshares/bitshares-ui/issues/1578#issuecomment-426741029

**Comment by startailcoon** [comment]
When can you push these changes @kapeer42? 

- We should commit this change without ANT updates as not everything is fixed to support it.
- Make sure the "Confirm Transaction" uses the propose toggle by default, and views the `create propose` instead of `transfer`. See flow below

This would behave the same way as legacy send, which would later be retired.

# Flow

## Without Propose
![image](https://user-images.githubusercontent.com/12114550/46435230-295ee580-c756-11e8-9d1a-ba4a06590d99.png)

Confirm signs transaction

## With Propose
On pushing "Propose" on send modal, skip directly to "Create Proposal" as "Propose From" is set in the modal.

![image](https://user-images.githubusercontent.com/12114550/46435495-be61de80-c756-11e8-9ef5-0889e6a804cb.png)

![image](https://user-images.githubusercontent.com/12114550/46435305-590ded80-c756-11e8-8152-31f936d4e64a.png)


# Flow Chart
- Send
  - Fill form as send
    - Sign `Transfer`
  - Fill form as propose
    - _Toggle `Propose` and select account and press `Propose`_ (skip this step)
    - Sign `Create Proposal`

**Comment by kapeer42** [comment]
@startailcoon Aye
